pilot/pkg/serviceregistry/kube/controller/endpoints.go

```diff
@@ -208,6 +208,17 @@ func (e *endpointsController) buildIstioEndpoints(endpoint interface{}, host hos
 	return endpoints
 }
 
+func (e *endpointsController) buildIstioEndpointsWithService(name, namespace string, host host.Name) []*model.IstioEndpoint {
+
+	ep, err := listerv1.NewEndpointsLister(e.informer.GetIndexer()).Endpoints(namespace).Get(name)
+	if err != nil || ep == nil {
```

Comment:
This might fail if endpoint does not exist when service was added which is the normal case? This is one of the reason I was thinking may be we should check in the controller before trying to build Endpoints of that service?

Reply:
isn't it fine if it fails? we just do nothing then, like before?

Reply:
Yes. Functionally it is fine. There wont be any issues except the below debug log would be misleading